IronFrame/ContainerHostService.cs

```diff
@@ -76,18 +76,18 @@ public IContainerHostClient StartContainerHost(string containerId, IContainerDir
                 Credentials = credentials,
             };
 
-            var hostProcess = processRunner.Run(hostRunSpec);
-
-            WaitForProcessToStart(hostProcess, HostProcessStartTimeout);
-
             // Order here is important.
-            // - Start the process and verify that it's running
+            // - Start the process 
             // - Add the process to the job object
+            // - Verify that the process is healthy
             // - Start the RPC message pump
             //
             // We need to ensure that the host process cannot create any new processes before
             // it's added to the job object.
+
+            var hostProcess = processRunner.Run(hostRunSpec);
             jobObject.AssignProcessToJob(hostProcess.Handle);
```

Comment:
My assumption is that the "important" comment is to make sure that everything else is done before "Start the RPC message pump". I looked around a bit more and I could not fine any issue by adding the process to the job object before checking if it is ready/healthy. If you see a concern, can you please walk me through a scenario step by step. 

I did change the order because if "WaitForProcessToStart(hostProcess, HostProcessStartTimeout)" fails or times out under load, the IronFrame.Host will never get terminated. But if it gets added in the job object first it will be tracked as soon as possible and get terminated at subsequent cleanup methods that will also terminate the job object. I also think that it is best to add a process as soon as possible in a job object.